src/System.Private.CoreLib/shared/System/Runtime/Serialization/SerializationInfo.cs

```diff
@@ -325,40 +342,51 @@ private object GetElementNoThrow(string name, out Type foundType)
                 return null;
             }
 
-            Debug.Assert(index < _values.Length, "[SerializationInfo.GetElement]index<m_data.Length");
-            Debug.Assert(index < _types.Length, "[SerializationInfo.GetElement]index<m_types.Length");
+            Debug.Assert(index < _values.Length, "[SerializationInfo.GetElement]index<_values.Length");
+            Debug.Assert(index < _types.Length, "[SerializationInfo.GetElement]index<_types.Length");
 
             foundType = _types[index];
-            Debug.Assert(foundType != null, "[SerializationInfo.GetElement]foundType!=null");
+            Debug.Assert((object)foundType != null, "[SerializationInfo.GetElement]foundType!=null");
             return _values[index];
         }
 
         public object GetValue(string name, Type type)
         {
-            if (type == null)
+            if ((object)type == null)
             {
                 throw new ArgumentNullException(nameof(type));
             }
 
+#if CORECLR
+            RuntimeType rt = type as RuntimeType;
```

Comment:
This check looks like a left-over from the partial trust support. I would remove it.

(In case you would like to keep it for some reason, change it to use `type.IsRuntimeImplemented()` without ifdef.)